crates/miden-objects/src/address/mod.rs

```diff
@@ -1,364 +1,223 @@
 mod r#type;
+use alloc::string::ToString;
+
+pub use r#type::AddressType;
+
+mod routing_parameters;
+use alloc::borrow::ToOwned;
+
+pub use routing_parameters::RoutingParameters;
 
 mod interface;
 mod network_id;
-use alloc::string::{String, ToString};
+use alloc::string::String;
 
-use bech32::Bech32m;
-use bech32::primitives::decode::{ByteIter, CheckedHrpstring};
 pub use interface::AddressInterface;
-use miden_core::utils::{ByteWriter, Deserializable, Serializable};
 use miden_processor::DeserializationError;
 pub use network_id::{CustomNetworkId, NetworkId};
-pub use r#type::AddressType;
 
 use crate::AddressError;
-use crate::account::{AccountId, AccountStorageMode};
-use crate::errors::Bech32Error;
+use crate::account::AccountStorageMode;
 use crate::note::NoteTag;
+use crate::utils::serde::{ByteWriter, Deserializable, Serializable};
+
+mod address_id;
+pub use address_id::AddressId;
 
 /// A user-facing address in Miden.
-#[non_exhaustive]
+///
+/// An address consists of an [`AddressId`] and optional [`RoutingParameters`].
+///
+/// A user who wants to receive a note creates an address and sends it to the sender of the note.
+/// The sender creates a note that only the address ID can consume (i.e. it provides access-control)
```

Comment:
The message gives the right idea, but I think it's too strong in its current wording.
Specifically, it need not be the case that only the address ID can consume the note (e.g. a P2ID note). It could be that the note is _consumable_ by anyone, but it is only _discoverable_ by the intended recipient.